cmd/onboard.go

```diff
@@ -187,65 +190,68 @@ TO1:
 			// A 25% plus or minus jitter is allowed by spec
 			select {
 			case <-ctx.Done():
-				return nil
+				return nil, ctx.Err()
 			case <-time.After(directive.Delay):
 			}
 		}
 	}
-	if to1d != nil {
-		for _, to2Addr := range to1d.Payload.Val.RV {
-			if to2Addr.DNSAddress == nil && to2Addr.IPAddress == nil {
-				slog.Error("Error: Both IP and DNS can't be null")
-				continue
-			}
+	if to1d == nil {
+		return nil, errors.Join(to1dErrs...)
+	}
+	for _, to2Addr := range to1d.Payload.Val.RV {
+		if to2Addr.DNSAddress == nil && to2Addr.IPAddress == nil {
+			slog.Error("Error: Both IP and DNS can't be null")
+			continue
```

Comment:
Error is logged here but not collected or returned. Can we do it similar to how its handled in line 183 `to1dErrs = append(to1dErrs, err)` ?

Reply:
added